web/backend/src/Server.ts

```diff
@@ -461,13 +475,18 @@ function sendToDela(dataStr: string, req: express.Request, res: express.Response
 
 // Secure /api/evoting to admins and operators
 app.use('/api/evoting/*', (req, res, next) => {
-  if (!isAuthorized(['admin', 'operator'], req)) {
-    console.log('role is:', req.session.role);
-    res.status(400).send('Unauthorized - only admins and operators allowed');
-    return;
-  }
-
-  next();
+  e.then((enforcer) => {
+    enforcer
+      .enforce(req.session.userid, 'election', 'create')
+      .then((isOk) => {
+        if (isOk) {
+          next();
+        } else {
+          res.status(400).send('Unauthorized - only admins and operators allowed');
+        }
+      })
+      .catch((error) => console.log('error', error));
+  }).catch((error1) => console.log('error', error1));
```

Comment:
As per our discussion today, this should work:
```suggestion
  e.then((enforcer) => {
    return enforcer
      .enforce(req.session.userid, 'election', 'create')
      .then((isOk) => {
        if (isOk) {
          next();
        } else {
          res.status(400).send('Unauthorized - only admins and operators allowed');
        }
      })
  }).catch((error) => console.log('error', error));
```

No need to do a double-catch 😁 
And since you're directly returning a value, you could also do this (notice the absence of `{}` and `return`):

```suggestion
  e.then((enforcer) =>
    enforcer
      .enforce(req.session.userid, 'election', 'create')
      .then((isOk) => {
        if (isOk) {
          next();
        } else {
          res.status(400).send('Unauthorized - only admins and operators allowed');
        }
      })).catch((error) => console.log('error', error));
```

Reply:
I modify that using the proposition of Noemien.